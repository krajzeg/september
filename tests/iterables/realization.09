original := [1, 2, 3, 4, 5, 6]

double := |x|{x * 2}
lessThan10 := |x|{x < 10}

chained := original.map(double).filter(lessThan10)
print("If chaining iterable operations works, this will be 2, 4, 6, 8:", chained)

realized := chained.realize()
print("If realization works, this will be 2, 4, 6, 8 too:", realized)
print("And the realized value will be an array, so this is True:", realized.is(Array))
