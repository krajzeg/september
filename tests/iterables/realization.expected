If chaining iterable operations works, this will be 2, 4, 6, 8: 2, 4, 6, 8
If realization works, this will be 2, 4, 6, 8 too: 2, 4, 6, 8
And the realized value will be an array, so this is True: <True>
